taskmanager/taskmanager.go

```diff
@@ -1,4 +1,161 @@
 /*
-Package taskmanager is the task/job management service layer for concurrent and asynchronous tasks
+Package taskmanager is the task/job management service layer for concurrent and asynchronous tasks with cron scheduling.
 */
 package taskmanager
+
+import (
+	"context"
+
+	"github.com/BuxOrg/bux/logging"
+	"github.com/newrelic/go-agent/v3/newrelic"
+	"github.com/robfig/cron/v3"
+	"github.com/rs/zerolog"
+	taskq "github.com/vmihailenco/taskq/v3"
+)
+
+type (
+
+	// TaskManager implements the Tasker interface
+	TaskManager struct {
+		options *options
+	}
+
+	// options holds all the configuration for the client
+	options struct {
+		cronService     *cron.Cron      // Internal cron job client
+		debug           bool            // For extra logs and additional debug information
+		logger          *zerolog.Logger // Internal logging
+		newRelicEnabled bool            // If NewRelic is enabled (parent application)
+		taskq           *taskqOptions   // All configuration and options for using TaskQ
+	}
+
+	// taskqOptions holds all the configuration for the TaskQ engine
+	taskqOptions struct {
+		config *taskq.QueueOptions    // Configuration for the TaskQ engine
+		queue  taskq.Queue            // Queue for TaskQ
+		tasks  map[string]*taskq.Task // Registered tasks
+	}
+)
+
+// NewTaskManager creates a new client for all TaskManager functionality
+// If no options are given, it will use local memory for the queue.
+// ctx may contain a NewRelic txn (or one will be created)
+func NewTaskManager(ctx context.Context, opts ...ClientOps) (Tasker, error) {
+	// Create a new tm with defaults
```

Comment:
Redundant comments

Reply:
I don't think so, because it says that by default, the local memory is used and I think it's work to notice that. About `ctx` comment, as I said, I left it as is, because it's related to `NewRelic feature`.